src/main/kotlin/no/nav/familie/ef/sak/vedtak/AndelHistorikkBeregner.kt

```diff
@@ -49,7 +50,36 @@ object AndelHistorikkBeregner {
 
     fun lagHistorikk(tilkjentYtelser: List<TilkjentYtelse>,
                      vedtaksliste: List<Vedtak>,
-                     behandlinger: List<Behandling>): List<AndelHistorikkDto> {
+                     behandlinger: List<Behandling>,
+                     tilOgMedBehandlingId: UUID?): List<AndelHistorikkDto> {
+        return if (tilOgMedBehandlingId == null) {
+            lagHistorikk(tilkjentYtelser, vedtaksliste, behandlinger)
+        } else {
+            lagHistorikkTilBehandlingId(tilkjentYtelser, vedtaksliste, behandlinger, tilOgMedBehandlingId)
+        }
+    }
+
+    /**
+     * Filtrerer vekk data som kommer etter behandlingen som man sender inn
+     */
+    private fun lagHistorikkTilBehandlingId(tilkjentYtelser: List<TilkjentYtelse>,
+                                            vedtaksliste: List<Vedtak>,
+                                            behandlinger: List<Behandling>,
+                                            tilOgMedBehandlingId: UUID?): List<AndelHistorikkDto> {
+        val filtrertBehandlinger = behandlinger.firstOrNull { it.id == tilOgMedBehandlingId }?.let { tomBehandling ->
```

Comment:
Kan vi trekke ut denne som egen konstant? 
`val tilOgMedBehandling = behandlinger.firstOrNull { it.id == tilOgMedBehandlingId } ?: error("Finner ikke behandling....")`

Reply:
Godt innspill, trakk ut denne i egen konstant og funksjon
f360b09